src/service-api.ts

```diff
@@ -183,6 +185,81 @@ const plugin: FastifyPluginAsync<GraaspImportZipPluginOptions> = async (fastify,
       return items;
     },
   );
+
+  // download item as zip
+  fastify.route<{ Params: { itemId: string } }>({
+    method: 'GET',
+    url: '/zip-export/:itemId',
+    schema: zipExport,
+    handler: async ({ member, params: { itemId }, log }, reply) => {
+      // get item info
+      const getItemTask = iTM.createGetTask(member, itemId);
+      const item = await runner.runSingle(getItemTask);
+
+      // init archive
+      const archive = archiver.create('zip', { store: true });
+      archive.on('warning', function (err) {
+        if (err.code === 'ENOENT') {
+          log.debug(err);
+        } else {
+          throw err;
+        }
+      });
+      archive.on('error', function (err) {
+        throw err;
+      });
+
+      // path to save files temporarly and save archive
+      const fileStorage = path.join(__dirname, TMP_FOLDER_PATH, item.id);
+      await mkdir(fileStorage, { recursive: true });
+      const zipPath = path.join(fileStorage, item.id + '.zip');
+      const zipStream = fs.createWriteStream(zipPath);
+      archive.pipe(zipStream);
+
+      // path used to index files in archive
+      const rootPath = path.dirname('./');
+
+      // create files from items
+      try {
+        await addItemToZip({
+          item,
+          archiveRootPath: rootPath,
+          archive,
+          member,
+          fileServiceType: SERVICE_ITEM_TYPE,
+          iTM,
+          runner,
+          fileTaskManager: fTM,
+          fileStorage,
+        });
+      } catch (error) {
+        throw new Error(`Error during exporting zip: ${error}`);
+      }
+
+      // wait for zip to be completely created
+      const sendBufferPromise = new Promise((resolve, reject) => {
+        zipStream.on('error', reject);
+
+        zipStream.on('close', () => {
+          // pipe archive data to the response
```

Comment:
Actually the comment should be rather `set reply headers depending zip file and return file`